core/src/main/java/org/apache/iceberg/jdbc/JdbcUtil.java

```diff
@@ -40,6 +40,8 @@ final class JdbcUtil {
   // property to control if view support is added to the existing database
   static final String SCHEMA_VERSION_PROPERTY = JdbcCatalog.PROPERTY_PREFIX + "schema-version";
 
+  static final String RETRYABLE_STATUS_CODES = "retryable_status_codes";
```

Comment:
should this have `JdbcCatalog.PROPERTY_PREFIX` as a prefix like the other properties have?

Reply:
I don't think so , looking at https://iceberg.apache.org/docs/nightly/jdbc/?h=jdbc#iceberg-jdbc-integration

The properties which are prefixed with "jdbc" look to be specifically for configuring the JDBC connection url, like user,password, ssl etc. These are standard options when establishing a JDBC connection. 

What we're trying to do here is independent of the JDBC connection URL and more like a typical catalog property.

I do see that we have strict-mode and schema-version catalog properties which kind of break this pattern since those aren't relevant for the JDBC connection URL, but I'm not sure we want to keep following that since it does seem like the original intention of the prefix was pretty clear on having it be for the connection URL. cc @jbonofre  if you have a take on this

Reply:
My take on this (maybe I'm wrong) is that anything specific to JDBC catalog should be prefixed by `jdbc`, just to be sure we don't have conflict with properties with same name not related to JDBC Catalog.
As JDBC Connection is used in JDBC Catalog, I would prefix with `jdbc` for consistency. As it's user facing properties, it's important to be consistent here.

Reply:
>  that anything specific to JDBC catalog should be prefixed by jdbc, just to be sure we don't have conflict with properties with same name not related to JDBC Catalog. As JDBC Connection is used in JDBC Catalog, I would prefix with jdbc for consistency. As it's user facing properties, it's important to be consistent here.

I think if we do this it's not really consistent with what we do with other catalogs. For example, REST, Glue, Nessie catalogs don't prefix any of their custom configurations with "glue", "rest", or "nessie" respectively; configuring
the catalog configuration is just a matter of setting the property via <catalog_name>.<property>. I also don't think we need to worry about conflicts in property names across different catalogs; in the end we use reflection to load the catalog and it's up to the catalog implementation to respect whichever properties we want, so we just need to make sure the properties are well defined.

The docs seemed pretty clear to me about the original intention of the "jdbc" prefix for the JDBC catalog https://iceberg.apache.org/docs/nightly/jdbc/?h=jdbc#configurations is that it's specifically meant for configuring JDBC connections (username/port/db name that are in the URL itself). 

`retryable_status_codes` is *not* configuring the JDBC connection URL. 
It is a catalog level property which controls when the catalog should retry establishing an entirely new connection so that's why I think it should be namespaced as it's own property (same as any other property that we do for other catalogs).

Reply:
Fair enough. It makes sense indeed. But it also means that `strict-mode` and `schema-version` should not have been prefixed by `jdbc` then.
We can change that, but as it's user facing, it could be problematic.
I can work on a PR to deprecated `jdbc` prefix (not still supporting it) for `strict-mode` and `schema-version`.

Reply:
>But it also means that strict-mode and schema-version should not have been prefixed by jdbc then.
Right, I think they should not have been. 

>We can change that, but as it's user facing, it could be problematic.
> I can work on a PR to deprecated jdbc prefix (not still supporting it) for strict-mode and schema-version

Sure, we can either deprecate the prefix or just leave as is. Since it's relatively early days for strict-mode/schema-version in the catalog (e.g. it'll practically take time for these paths to get exercised, as views get adopted) I think it makes most sense just to deprecate and correct it.